master/buildbot/util/steps.py

```diff
@@ -0,0 +1,94 @@
+# This file is part of Buildbot.  Buildbot is free software: you can
+# redistribute it and/or modify it under the terms of the GNU General Public
+# License as published by the Free Software Foundation, version 2.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+# details.
+#
+# You should have received a copy of the GNU General Public License along with
+# this program; if not, write to the Free Software Foundation, Inc., 51
+# Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+# Copyright Buildbot Team Members
+import urllib
+
+from twisted.internet import defer
+
+from buildbot import util
+from buildbot.status.web.base import css_classes
+from buildbot.status.web.base import path_to_step
+
+
+@defer.inlineCallbacks
+def get_steps(steps_list, codebases_arg, request):
+    """ This function return steps list with full description
+
+    :param steps_list: list of steps in build
+    :param codebases_arg: additional parameters for url
+    :param request: http request object
+    :return:
+    """
+    steps = []
+    for step in steps_list:
+        step_obj = {
+            'name': step.getName(),
+            'css_class': 'not-started',
+            'time_to_run': '',
+            'link': path_to_step(request, step),
+            'text': ' '.join(step.getText()),
+            'urls': [],
+            'logs': [],
+        }
+        if step.isFinished and step.isHidden():
```

Comment:
AFAIK step.isFinished is method, not field, so `step.isFinished() and step.isHidden()`